### PR TITLE
existingMeta = ignore for applyChangeMeta

### PR DESCRIPTION
**Pull request by beckerbenj**
This adresses #25. 

Unfortunately the `recode_labels()` source code is rather messy. Cleaning it up (probably splitting into separate sub-functions) would be desirable but is beyond the scope of this PR.

### CODE REVIEW

File: tests/testthat/test_applyChangeMeta.R [hunk 1]
```diff
@@ -180,14 +188,14 @@ test_that("Expand labels", {
...
-  out <- recode_labels(dfSAV$labels, changes_val2)
+  out <- recode_labels(dfSAV$labels, changes_val2, existingMeta = "stop")
   expect_equal(out$value[1:4], c(-99, -96, 1, 2))
   expect_equal(dim(out), c(8, 8))
 
   df1_changes <- getChangeMeta(df1, level = "value")
   df1_changes[2, "value_new"] <- 99
   df1_changes[2, "valLabel_new"] <- "test"
-  out2 <- recode_labels(df1$labels, df1_changes)
+  out2 <- recode_labels(df1$labels, df1_changes, existingMeta = "stop")
```
Comment (franikowsp) [inline tests/testthat/test_applyChangeMeta.R:198]: LGTM.

File: R/applyChangeMeta.R [hunk 2]
```diff
@@ -197,16 +197,23 @@ recode_labels <- function(labels, changeTable, existingMeta) {
     dup_recode_values <- dup_recode_values[!dup_recode_values %in% single_simpleChanges[existing_value_vec, "value"]]
     new_dup_value_vec <- !is.na(single_simpleChanges$value_new) & single_simpleChanges$value_new %in% dup_recode_values
 
+    # ignore behavior: reset all changes for duplicate and existing value labels & missing tags
+    if(identical(existingMeta, "ignore")) {
+      single_simpleChanges[new_dup_value_vec | existing_value_vec, "value_new"] <- NA
+    }
+
```
Comment (franikowsp) [inline R/applyChangeMeta.R:204]: LGTM

File: R/applyChangeMeta.R [hunk 3]
```diff
@@ -197,16 +197,23 @@ recode_labels <- function(labels, changeTable, existingMeta) {
...
       }
       # drop behavior
-      remove_value_meta <- single_simpleChanges[new_dup_value_vec, "value"]
-      drop_meta_rows <- which(single_labels$value %in% remove_value_meta)
-      single_labels[drop_meta_rows, c("valLabel", "missings")] <- NA
+      if(identical(existingMeta, "drop")) {
+        remove_value_meta <- single_simpleChanges[new_dup_value_vec, "value"]
+        drop_meta_rows <- which(single_labels$value %in% remove_value_meta)
+        single_labels[drop_meta_rows, c("valLabel", "missings")] <- NA
+      }
```
Comment (franikowsp) [inline R/applyChangeMeta.R:216]: LGTM

File: R/applyChangeMeta.R [hunk 4]
```diff
@@ -197,16 +197,23 @@ recode_labels <- function(labels, changeTable, existingMeta) {
...
     new_dup_value_vec <- !is.na(single_simpleChanges$value_new) & single_simpleChanges$value_new %in% dup_recode_values
 
+    # ignore behavior: reset all changes for duplicate and existing value labels & missing tags
+    if(identical(existingMeta, "ignore")) {
+      single_simpleChanges[new_dup_value_vec | existing_value_vec, "value_new"] <- NA
+    }
+
     if(length(dup_recode_values) > 0 && !any(existing_value_vec)){
-      if(!identical(existingMeta, "drop")) {
+      if(!(identical(existingMeta, "drop") || identical(existingMeta, "ignore"))) {
```
Comment (franikowsp) [inline R/applyChangeMeta.R:206]: LGTM

**Review by franikowsp — APPROVED** [review]
Looks all good to me, again (note: I only found the time for a "high-level review" as the function and its behavior are already quite complex 🙈)!

File: R/applyChangeMeta.R [hunk 5]
```diff
@@ -11,7 +11,8 @@
 #' data in the \code{changeTable} or, if incomplete, from the recoded value (\code{"value_new"}).
 #'
 #' Furthermore, one might recode multiple old values in the same new value. This is currently only possible with
-#' \code{existingMeta = "drop"}, which drops all related meta data on value level.
+#' \code{existingMeta = "drop"}, which drops all related meta data on value level or
+#' \code{existingMeta = "ignore"}, which leaves all related meta data on value level untouched.
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply R/applyChangeMeta.R:15]: Agree, fixed!

File: tests/testthat/test_applyChangeMeta.R [hunk 6]
```diff
@@ -98,6 +98,7 @@ test_that("Recoding with value meta data conflicts", {
 test_that("Recoding multiple value into the same value (without meta data conflicts)", {
   changes_val2 <- changes_val
   changes_val2[1:3, "value_new"] <- 10
+  changes_val4 <- changes_val3 <- changes_val2
   expect_error(out <- recode_labels(dfSAV$labels, changes_val2, existingMeta = "stop"),
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply tests/testthat/test_applyChangeMeta.R:102]: Agree and fixed!

File: tests/testthat/test_applyChangeMeta.R [hunk 7]
```diff
@@ -98,6 +98,7 @@ test_that("Recoding with value meta data conflicts", {
 test_that("Recoding multiple value into the same value (without meta data conflicts)", {
   changes_val2 <- changes_val
   changes_val2[1:3, "value_new"] <- 10
+  changes_val4 <- changes_val3 <- changes_val2
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply tests/testthat/test_applyChangeMeta.R:101]: Removed!

File: tests/testthat/test_applyChangeMeta.R [hunk 8]
```diff
@@ -110,6 +111,13 @@ test_that("Recoding multiple value into the same value (without meta data confli
   expect_equal(out2[1, "valLabel"], NA_character_)
   expect_equal(out2[1, "missings"], "valid")
   expect_equal(out2[2, "varName"], "VAR2")
+
+  out3 <- recode_labels(dfSAV$labels, changes_val2, existingMeta = "ignore")
+  expect_equal(extractMeta(out3, "VAR1"), extractMeta(dfSAV, "VAR1"))
+
+  changes_val3[1:3, "value_new"] <- c(1, 1, NA)
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply tests/testthat/test_applyChangeMeta.R:118]: Relocated as suggested.

**Comment by beckerbenj** [comment]
This closes #25

**Pull request merged**